File: 409_Longest_Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> d;
        for (int i = 0; i < s.size(); i++) {
            d[s[i]]++;
        }
        int result = 0;
        bool odd = true;
        for (int i = 0; i < d.size(); i++) {
            if (d[i] % 2 == 1) {
                result += d[i] - 1;
                if (odd) {
                    result++;
                    odd = false;
                }
            }
            else {
                result += d[i];
            }
        }
        return result;
    }
};

File: 409_Longest_Palindrome.py
from collections import Counter
class Solution:
    def longestPalindrome(self, s: str) -> int:
        gotOdd = False
        ret = 0
        for _, v in Counter(s).items():
            if v % 2 == 0:
                ret += v
            else:
                ret += v - 1
                if not gotOdd:
                    ret += 1
                    gotOdd = True
        
        return ret
